File: ui/src/types/sports.ts
import type { LogEntry } from './shared'

export interface SportsSignal {
  market_id: string
  home: string
  away: string
  date: string
  side: string
  edge_pct: number
  kelly_size: number
  strategy_id: string
  status: string
}

export interface SportsStrategy {
  id: string
  name: string
  description: string
  enabled: boolean
  auto_execute: boolean
}

export interface Fixture {
  home: string
  away: string
  date: string
  home_xg: number
  away_xg: number
  polymarket_id: string | null
}

export interface LiveMatch {
  home_team: string
  away_team: string
  home_goals: number
  away_goals: number
  minute: number
}

export interface SportsState {
  strategies: SportsStrategy[]
  signals: SportsSignal[]
  fixtures: Fixture[]
  live_matches: LiveMatch[]
  logs: LogEntry[]
}

File: src/commands/dto/sports.rs
//! Sports tab DTOs.

use serde::Serialize;

#[derive(Serialize, Clone)]
pub struct SportsSignalDto {
    pub market_id: String,
    pub home: String,
    pub away: String,
    pub date: String,
    pub side: String,
    pub edge_pct: f64,
    pub kelly_size: f64,
    pub strategy_id: String,
    pub status: String,
}

#[derive(Serialize, Clone)]
pub struct SportsStrategyDto {
    pub id: String,
    pub name: String,
    pub description: String,
    pub enabled: bool,
    pub auto_execute: bool,
}

#[derive(Serialize, Clone)]
pub struct FixtureDto {
    pub home: String,
    pub away: String,
    pub date: String,
    pub home_xg: f64,
    pub away_xg: f64,
    pub polymarket_id: Option<String>,
}

#[derive(Serialize, Clone)]
pub struct LiveMatchDto {
    pub home_team: String,
    pub away_team: String,
    pub home_goals: u8,
    pub away_goals: u8,
    pub minute: u8,
}

#[derive(Serialize, Clone)]
pub struct SportsStateDto {
    pub strategies: Vec<SportsStrategyDto>,
    pub signals: Vec<SportsSignalDto>,
    pub fixtures: Vec<FixtureDto>,
    pub live_matches: Vec<LiveMatchDto>,
    pub logs: Vec<super::shared::LogEntryDto>,
}
